### PR TITLE
Remove old go build constraints

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Removes the <1.17 build constraints. It adds an entry to `.git-blame-ignore-revs` file to keep it out of blames. 

**THIS PR SHOULD NOT BE SQUASH MERGED**

### Motivation

They are no longer needed since we are 2+ versions away from when they were used.

### Additional Notes

It is probably easier to review the non-fix commits separately:
https://github.com/DataDog/datadog-agent/pull/16181/commits/303f3cb5dfe192e08f76d614ae3c6ea53aeff911
https://github.com/DataDog/datadog-agent/pull/16181/commits/c148c8bda25f19dec32f6e5fcdd61ddcf49d1402

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

**Review by guyarb — APPROVED** [review]
LGTM for USM files

**Review by vickenty — APPROVED** [review]
LGTM for metrics-logs.

**Review by ahmed-mez — APPROVED** [review]

**Review by maycmlee — APPROVED** [review]
👍 for docs

**Review by hmahmood — APPROVED** [review]

**Review by sgnn7 — APPROVED** [review]

File: pkg/clusteragent/custommetrics/api/generated/openapi/zz_generated.openapi.go [hunk 1]
```diff
@@ -1,5 +1,4 @@
 //go:build !ignore_autogenerated
-// +build !ignore_autogenerated
```
[the remark this answers is not in the document]
Comment (sgnn7) [reply pkg/clusteragent/custommetrics/api/generated/openapi/zz_generated.openapi.go:2]: No clue tbh - I don't know what OpenAPI generator does with these but I would assume it includes them for portability 🤷

**Review by robertjli — APPROVED** [review]

**Review by cedricvanrompay-datadog — APPROVED** [review]

**Review by usamasaqib — APPROVED** [review]

**Pull request merged**